llvm/lib/CAS/OnDiskGraphDB.cpp

```diff
@@ -1280,21 +1306,27 @@ OnDiskGraphDB::open(StringRef AbsPath, StringRef HashName,
     return std::move(E);
 
   std::optional<OnDiskDataAllocator> DataPool;
+  StringRef PolicyName =
+      Policy == FaultInPolicy::SingleNode ? "single" : "full";
   if (Error E = OnDiskDataAllocator::create(
                     AbsPath + Slash + FilePrefix + DataPoolFile,
-                    DataPoolTableName + "[" + HashName + "]",
+                    DataPoolTableName + "[" + HashName + "]" + PolicyName,
```

Comment:
Maybe `PolicyName` goes into `OnDiskHashMappedTrie` path as well?

Reply:
I don't think there's any benefit, these are always opened together.

Reply:
Well, technically, you can create two DB at the same location with different fault in policy but share the Trie?

Reply:
> you can create two DB at the same directory with different fault in policy but share the Trie?

You can't do that with how `OnDiskGraphDB` is implemented, do you mean that a future implementation of `OnDiskGraphDB` may want to do that?

Reply:
I actually don't want that to happen but I don't see how you can prevent that with current implementation.

Reply:
It will be an error if you try to create a DB on the same directory with different policy. Unit test `TEST(OnDiskGraphDBTest, FaultInPolicyConflict)` checks for that.